types/jsonwebtoken/index.d.ts

```diff
@@ -169,7 +169,7 @@ export function sign(
  * [options] - Options for the verification
  * returns - The decoded token.
  */
-export function verify(token: string, secretOrPublicKey: Secret, options?: VerifyOptions): object | string;
+export function verify<T>(token: string, secretOrPublicKey: Secret, options?: VerifyOptions): T | string;
```

Comment:
This probably should read:
```suggestion
export function verify<T = object>(token: string, secretOrPublicKey: Secret, options?: VerifyOptions): T | string;
```
to stay in sync with previous implementation (backward comp.):
```ts
jwt.verify(token, cert, { audience: [/urn:f[o]{2}/, "urn:bar"] }, (err, decoded) => {
    // if audience mismatch, err == invalid audience
    const result = decoded as TestObject;
});
```
while providing all benefits of generics in a rewritten test case:
```ts
jwt.verify<TestObject>(token, getKey, (err, decoded) => {
    const result = decoded;
    console.log(result.foo); // bar
});
```

Reply:
Hmmm, shouldn't this be `T extends object`? I don't think that the library can return anything else than an object in that place (e.g. `number`, `boolean`, etc.)